### ISSUE TITLE
createStatisticsSnapshot not always reliable 

### ISSUE THREAD

**Issue by jeggy** [nidomiro/KDataLoader#8]
As seen here: ~~https://github.com/nidomiro/KDataLoader/commit/649109a3a710f26cdb19406e35206c849d7bbc60/checks?check_suite_id=410490317 Link somehow changed.~~
a Job failed once at `DataLoaderStatisticsTest.kt:242`.

I did also have this issue when creating my [latest PR(#7)](https://github.com/nidomiro/KDataLoader/pull/7), which I solved by doing a hack with using `delay`:
https://github.com/nidomiro/KDataLoader/blob/master/src/jvmTest/kotlin/nidomiro/kdataloader/DSLTest.kt#L161-L185

One solution could be these changes in `SimpleDataLoaderImpl`
```kotlin
statisticsScope = CoroutineScope(Dispatchers.Default)
// and add this:
private val statisticsJobs = mutableListOf<Deferred<Unit>>()

// ... and
return if (options.cacheEnabled) {
    options.cache.getOrCreate(key, block, { statisticsScope.launch { statisticsCollector.incCacheHitCount() } })
    // Changed to :
    options.cache.getOrCreate(key, block, { statisticsJobs.add(statisticsScope.async { statisticsCollector.incCacheHitCount(); Unit }) })
} else {
    block(key)
}

// ... and then
override suspend fun createStatisticsSnapshot(): DataLoaderStatistics {
    statisticsJobs.awaitAll()
    return statisticsCollector.createStatisticsSnapshot()
}
```

**Comment by nidomiro** [comment]
Thanks for the Report 👍 
I had some random test-fails and I think it's caused by the same error.
Your solution looks very good for me 👍
I will implement it like this

**Comment by nidomiro** [comment]
@jeggy it's fixed in 0.2.0
I switched the project to `Multiplatform`, so the ArtifactId changed from `KDataLoader` to `KDataLoader-jvm` (for a `jvm`-build)
